### PR TITLE
Code refactoring for the reading from ETH boards

### PR DESCRIPTION
**Pull request by marcoaccame**
The blocking reading calls sent directly to ETH boards are implemented in a more efficient and flexible way. Every ask<> message has a dedicated signature which is used inside the reply say<> message to release a blocking semaphore. This is the first step towards a the use of parallel ask<> messages sent to one or more ETH boards. This feature will be used to accelerate the reading calls done to a body part of a speeding factor equal the number of joints.

This PR contains changes almost confined to src\libraries\icubmod\embObjLib foledr, does not contain behavioral changes of the devices and has been extensively tested on the iCubGenova04 robot all through its development.

### CODE REVIEW

File: src/libraries/icubmod/embObjMotionControl/embObjMotionControl.cpp [hunk 1]
```diff
@@ -1542,19 +1505,10 @@ bool embObjMotionControl::init()
         }
     }
 
-    Time::delay(0.005);  // 5 ms
-
-
+    Time::delay(0.005);
```
Comment (barbalberto) [inline src/libraries/icubmod/embObjMotionControl/embObjMotionControl.cpp]: Please use `SystemClock::delaySystem()`

File: src/libraries/icubmod/embObjStrain/embObjStrain.cpp [hunk 2]
```diff
@@ -449,7 +371,7 @@ bool embObjStrain::fillScaleFactor()
     // wait for response
     while(!gotFullScaleValues && (timeout != 0))
     {
-        res->addSetMessage(id32_strain_config, (uint8_t *) &strainConfig);
+        res->setRemoteValue(id32_strain_config, &strainConfig);
         Time::delay(1.0);
```
Comment (barbalberto) [inline src/libraries/icubmod/embObjStrain/embObjStrain.cpp]: Use `SystemClock::delaySystem()`

Comment (marcoaccame) [reply src/libraries/icubmod/embObjStrain/embObjStrain.cpp]: ok, i have used SystemClock::delaySystem() and SystemClock::nowSystem()

File: src/libraries/icubmod/embObjLib/FeatureInterface.cpp [hunk 3]
```diff
@@ -26,6 +26,9 @@
 #include <ace/config.h>
 #include <ace/Recursive_Thread_Mutex.h>
 
+#include <theNVmanager.h>
+using namespace tbd;
```
[the remark this answers is not in the document]
Comment (marcoaccame) [reply src/libraries/icubmod/embObjLib/FeatureInterface.cpp]: used eth::

**Comment by barbalberto** [comment]
Linux tests are passed, Mac one are not running for some reason, but this code is not supposed to run on mac, so I merge it.

**Comment by traversaro** [comment]
@barbalberto Nitpick: the `embObj*` libraries/devices are not compiled by Travis, neither on Linux or on macOS .

**Pull request merged**